bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -209,35 +251,59 @@ protected NeoHubInfoResponse fromNeoHubReadInfoResponse() {
                 updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
             }
 
-            return newInfoResponse;
+            if (deviceData instanceof NeoHubLiveDeviceData) {
+                long temp = ((NeoHubLiveDeviceData) deviceData).getTimestampSystem();
+                if (temp != systemTimestamp) {
+                    systemDataDirty = true;
+                    systemTimestamp = temp;
+                }
+            } else {
+                if (Instant.now().isAfter(systemLastRefreshed.plusSeconds(3600))) {
+                    systemDataDirty = true;
+                    systemLastRefreshed = Instant.now();
+                }
+            }
```

Comment:
The hub provides two sorts of data: the fast changing live data from devices, and the slow changing settings from users. The fast changing data is called `Info` in the old Api and `LiveData` in the new Api. And the slow changing data is called `DCB` (old) and `System` (new). The fast changing data is polled (according to pollInterval) every 60 seconds. And the slow changing data is read less frequently. In the new Api, the `LiveData` includes a time stamp tag on the `System` data so the data can be automatically read (only) when the time stamp changes (systemDirty). However in the old Api the `Info` does not include such a time stamp, so instead the binding has to “assume” that the data is dirty from time to time, which it does every one hour (even if probably the data has not changed).

Perhaps there is a better way to handle the polling of the slow changing data? In particular with the old Api? Do you have some suggestions?

You are partly right about the split of logic between the handler and the getters; the getters do indeed read the actual data, but only the handler knows the context about what changed (time stamp tag) since the getter was last called. Nevertheless I will see if I can shift some of the logic to the getters whilst still keeping the context in the handler.

Reply:
Hmm. I have been thinking some more about this. And I have made some changes about how the timestamps are checked (if any), and how systemData is updated. I hope it is Ok with you.